File: src/components/pages/projects/$projectId/shared/connect-snippets/flutter/lib/sign_in.dart
import 'package:flutter/material.dart';
import 'package:appwrite/appwrite.dart';
import 'appwrite_client.dart';

class SignIn extends StatefulWidget {
  const SignIn({
    super.key,
    required this.onSignedIn,
    required this.onGoToSignUp,
  });

  final VoidCallback onSignedIn;
  final VoidCallback onGoToSignUp;

  @override
  State<SignIn> createState() => _SignInState();
}

class _SignInState extends State<SignIn> {
  final email = TextEditingController();
  final password = TextEditingController();
  String error = '';

  @override
  void dispose() {
    email.dispose();
    password.dispose();
    super.dispose();
  }

  Future<void> _submit() async {
    if (email.text.isEmpty || password.text.isEmpty) return;
    setState(() => error = '');
    try {
      await Account(client).createEmailPasswordSession(
        email: email.text,
        password: password.text,
      );
      widget.onSignedIn();
    } on AppwriteException catch (e) {
      setState(() => error = e.message ?? 'Sign in failed');
    } catch (_) {
      setState(() => error = 'Sign in failed');
    }
  }

  @override
  Widget build(BuildContext context) {
    return ListView(
      padding: const EdgeInsets.all(24),
      children: [
        const Text('Sign in'),
        if (error.isNotEmpty) Text(error),
        TextField(
          controller: email,
          keyboardType: TextInputType.emailAddress,
          autocorrect: false,
          decoration: const InputDecoration(hintText: 'Email'),
        ),
        TextField(
          controller: password,
          obscureText: true,
          decoration: const InputDecoration(hintText: 'Password'),
        ),
        TextButton(onPressed: _submit, child: const Text('Sign in')),
        Row(
          children: [
            const Text('No account? '),
            TextButton(
              onPressed: widget.onGoToSignUp,
              child: const Text('Sign up'),
            ),
          ],
        ),
      ],
    );
  }
}

File: src/components/pages/projects/$projectId/shared/connect-snippets/web/react/cra/src/pages/SignIn.js
import { useState } from 'react'
import { Account } from 'appwrite'
import { client } from '../lib/appwrite'

export function SignIn() {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [error, setError] = useState('')

  async function handleSubmit(e) {
    e.preventDefault()
    setError('')
    try {
      const account = new Account(client)
      await account.createEmailPasswordSession({ email, password })
      window.location.href = '/'
    } catch (err) {
      setError(err?.message || 'Sign in failed')
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <h1>Sign in</h1>
      {error ? <p>{error}</p> : null}
      <input
        type="email"
        placeholder="Email"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
        required
      />
      <input
        type="password"
        placeholder="Password"
        value={password}
        onChange={(e) => setPassword(e.target.value)}
        required
      />
      <button type="submit">Sign in</button>
      <p>
        No account? <a href="/sign-up">Sign up</a>
      </p>
    </form>
  )
}
